setup.py

```diff
@@ -30,8 +32,83 @@
 from distutils.command.build_scripts import build_scripts as BuildScripts
 from distutils.command.sdist import sdist as SDist
 
-sys.path.insert(0, os.path.abspath('lib'))
-from ansible.release import __version__, __author__
+
+def find_package_info(*file_paths):
+    # Open in Latin-1 so that we avoid encoding errors.
+    # Use codecs.open for Python 2 compatibility
+    try:
+        with codecs.open(os.path.join(*file_paths), 'r', 'latin1') as f:
```

Comment:
Latin1 seems wrong...
* In python 3, the default encoding of python files is utf8. 
* When we explicitly designate for python 2, we use utf-8

If you don't want to trust that this is utf-8 (but i think we should), then we should use bytes to read and convert that to text at some point before returning.

Reply:
That is true but also the target file only contains ASCII so this will not fail as long as we don't "add emojis" there at which point it'll fail in the CI. So there's a really low risk of shipping broken dists...

Reply:
This came directly from pypa, so I'm just using their recommendation.  Their version has changed in recent years where py2 doesn't matter.  But I didn't write any of this.

Reply:
Yeah, that's from 2013. Nowadays it probably matters less. If only we had less weird things in setup.py, we could just migrate to setup.cfg mostly...

Reply:
Here is the current version of that recommendation: https://packaging.python.org/guides/single-sourcing-package-version/

Reply:
<nod> okay,.  Since we control the release.py  file, we probably shouldn't add this particular encoding code, then.  It overcomplicates things and confused people who don't have encoding experience won't understand why latin1 is used here and will assume the file actually contains characters encoded in Latin 1). I suggest either ignoring the encoding question if we are going to say the release.py file is ascii-only or reading as bytes and then decoding from utf8 if we want to future proof it against things like a non-ascii code name. (Importing and using codecs is another thing that will be nonobvious to people who are unfamiliar with encoding issues)

Reply:
(the pypa recommendation is written to work with any project under any conditions which makes it overly complex for most projects actually implementing it.)